File: backend/goat_listing/src/main/java/com/goat/cart/domain/exceptions/ItemAlreadyInCartException.java
package com.goat.cart.domain.exceptions;

/**
 * Excepción lanzada cuando se intenta agregar un item que ya existe en el carrito.
 */
public class ItemAlreadyInCartException extends RuntimeException {
    public ItemAlreadyInCartException(String message) {
        super(message);
    }

    public ItemAlreadyInCartException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: backend/goat_catalog/src/goat_catalog/cart/domain/exceptions/item_already_in_cart_exception.py
"""Excepción cuando un item ya está en el carrito."""


class ItemAlreadyInCartException(Exception):
    """Excepción lanzada cuando se intenta agregar un item que ya existe."""

    def __init__(self, message: str = "El item ya está en el carrito") -> None:
        """Inicializa la excepción.

        Args:
            message: Mensaje de error
        """
        super().__init__(message)
        self.message = message

